lib/matplotlib/cbook/__init__.py

```diff
@@ -2231,3 +2223,20 @@ def _picklable_class_constructor(mixin_class, fmt, attr_name, base_class):
     factory = _make_class_factory(mixin_class, fmt, attr_name)
     cls = factory(base_class)
     return cls.__new__(cls)
+
+
+def _unpack_to_numpy(x):
+    """Internal helper to extract data from e.g. pandas and xarray objects."""
+    if isinstance(x, np.ndarray):
+        # If numpy, return directly
+        return x
+    if hasattr(x, 'to_numpy'):
+        # Assume that any function to_numpy() do actually return a numpy array
+        return x.to_numpy()
+    if hasattr(x, 'values'):
```

Comment:
I think you can simplify the nested ifs and temporary array here?

```suggestion
    if isinstance(getattr(x, 'values', None), np.ndarray):
        return x.values
```

Reply:
Oh, I think the idea was to avoid calling values several times in case the generation was non-trivial.

Reply:
Good point, thanks for clarifying!